File: backend/app/ai/tools/services.py
from typing import Optional, Dict, Any
from sqlalchemy.orm import Session

from app.models.service import Service


def get_service_tool(db: Session, community_id: str, query: str) -> Optional[Dict[str, Any]]:
    """Retrieve institutional service record with hours and contacts."""
    clean_q = query.lower().strip()

    services = (
        db.query(Service)
        .filter(
            Service.community_id == community_id,
            Service.is_active.is_(True),
        )
        .all()
    )

    matched = None
    for s in services:
        if clean_q in s.name.lower() or s.name.lower() in clean_q or clean_q in s.department.lower():
            matched = s
            break

    if not matched:
        return None

    hours_desc = "9:00 AM – 5:00 PM (Mon – Fri)"
    if matched.hours:
        hours_desc = "Regular operating hours"

    return {
        "id": matched.id,
        "name": matched.name,
        "description": matched.description,
        "category": matched.category or "Student Services",
        "department": matched.department,
        "location": matched.location_id or "Student Services Center (SJT-G12)",
        "hours": hours_desc,
        "contactEmail": matched.contact or "support@nexora.edu",
        "isUrgent": matched.is_urgent,
    }

File: lib/api/services.ts
import { apiRequest, ApiResponse, createSuccessResponse, createErrorResponse } from './client';
import { ServiceItem } from '../types';
import { mockServices } from '../mock-data';

function mapBackendService(s: any): ServiceItem {
  return {
    id: s.id,
    name: s.name,
    description: s.description,
    category: s.category || 'General',
    department: s.department,
    location: s.location_id || 'Student Center',
    hours: s.hours && s.hours.length > 0 ? 'Mon-Fri 09:00 - 17:00' : '09:00 AM - 05:00 PM',
    contactEmail: s.contact || undefined,
    isUrgent: s.is_urgent ?? false,
  };
}

export const servicesApi = {
  async getServices(category?: string): Promise<ApiResponse<ServiceItem[]>> {
    const params = new URLSearchParams();
    if (category && category !== 'All') params.append('category', category);

    const queryString = params.toString() ? `?${params.toString()}` : '';
    const res = await apiRequest<any[]>(`/api/v1/services${queryString}`);

    if (res.data && Array.isArray(res.data) && res.data.length > 0) {
      return {
        data: res.data.map(mapBackendService),
        error: null,
        status: 200,
      };
    }

    if (!category || category === 'All') {
      return createSuccessResponse<ServiceItem[]>(mockServices);
    }
    const filtered = mockServices.filter(
      (s) => s.category.toLowerCase() === category.toLowerCase()
    );
    return createSuccessResponse<ServiceItem[]>(filtered);
  },

  async getServiceById(id: string): Promise<ApiResponse<ServiceItem>> {
    const res = await apiRequest<any>(`/api/v1/services/${id}`);
    if (res.data) {
      return {
        data: mapBackendService(res.data),
        error: null,
        status: 200,
      };
    }

    const item = mockServices.find((s) => s.id === id);
    if (!item) {
      return createErrorResponse<ServiceItem>('Service not found', 404);
    }
    return createSuccessResponse<ServiceItem>(item);
  },
};
